### PR TITLE
[MA] Updating models Ctrs to take required args

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/MetricAnomalyAlertSnoozeCondition.java [hunk 1]
```diff
@@ -1,89 +1,90 @@
...
-     * Set the snoozeScope property: snooze scope.
+     * Sets the value indicating the snooze point count, the value range is : [0, +∞).
      *
-     * @param snoozeScope the snoozeScope value to set.
+     * @param autoSnooze the the snooze point count.
      * @return the MetricAnomalyAlertSnoozeCondition object itself.
      */
-    public MetricAnomalyAlertSnoozeCondition setSnoozeScope(SnoozeScope snoozeScope) {
-        this.snoozeScope = snoozeScope;
+    public MetricAnomalyAlertSnoozeCondition setAutoSnooze(Integer autoSnooze) {
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/MetricAnomalyAlertSnoozeCondition.java:65]: Should this property also have a setter in spite of being a required param in the constructor?

**Review by samvaity — APPROVED** [review]
💯 :shipit:

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/MetricAnomalyAlertSnoozeCondition.java (hunk 1 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/MetricAnomalyAlertSnoozeCondition.java:65]: Discussed offline; this is a shared model for create|get|update APIs hence we expose setters to enable update use-cases.

**Pull request merged**